### PR TITLE
feat(web): prevent blank map on mobile apps

### PR DESCRIPTION
**Pull request by tonypls**
## Issue
Opening the mobile app after some time results in a blank map

## Description
This PR adds a check to see if the app is being opened from the background and reloads the map is the sources are not present.

Would love feedback if there's a better way to do this.

### CODE REVIEW

File: web/src/features/map/Map.tsx [hunk 1]
```diff
@@ -79,10 +81,40 @@ export default function MapPage({ onMapLoad }: MapPageProps): ReactElement {
...
     setMapReference(reference);
   }, []);
 
+  useEffect(() => {
+    let subscription: PluginListenerHandle | null = null;
+
+    const setupListener = async () => {
+      subscription = await App.addListener('appStateChange', ({ isActive }) => {
+        if (!isActive) {
+          setWasInBackground(true);
```
[the remark this answers is not in the document]
Comment (VIKTORVAV99) [reply web/src/features/map/Map.tsx:95]: I think the AI is wrong on this one but I'm not sure...

**Comment by tonypls** [comment]
> Is there a way to test this without like waiting for hours for this to actually happen on a real device?
> 
> Code looks good and all but I can't really tell if it actually works...

Good idea, I've added a temporary mapKill function, type mapKill() in the console then you can test the re-focussing

**Comment by VIKTORVAV99** [comment]
> > Is there a way to test this without like waiting for hours for this to actually happen on a real device?
> > Code looks good and all but I can't really tell if it actually works...
> 
> Good idea, I've added a temporary mapKill function, type mapKill() in the console then you can test the re-focussing

Great! Just tried it and it works great between tab switches at least. And it looks like it will work in the app as well but I don't think there is a way to bring up the console there.

Might I suggest we keep the kill function but put it behind a (env.DEV) flag so it's stripped from production builds?

**Review by VIKTORVAV99 — APPROVED** [review]
LGTM! 🎉

**Pull request merged**